### DISCUSSION TITLE
Does bitECS take advantage of WebWorkers?

### DISCUSSION THREAD

**Discussion by adamwong246** [NateTheGreatt/bitECS#158] [General]
I am gathering tools for my next project and I am hoping the bitECS can fill a role. However, I would like to take advantage of webworkers. Can bitECS provide that functionality?

**Comment by adamwong246** [comment]
I did some digging and found that underlaying bitECS is Sets and Maps, which are not transferrable or shareable. Such would be a massive re-write. I think perhaps the best solution maybe to write to a sharedarraybuffer in the main iteration loop, and then, on the rendering thread, read from that shared buffer.

**Comment by NateTheGreatt** [comment]
i wrote some threading strategies for the new v0.4 version of bitecs here:
https://github.com/NateTheGreatt/bitECS/blob/rc-0-4-0/docs/Multithreading.md
